rasterio/transform.py

```diff
@@ -267,3 +250,192 @@ def from_gcps(gcps):
 
     """
     return Affine.from_gdal(*_transform_from_gcps(gcps))
+
+
+def affine_to_ndarray(transform):
+    """
+    Create ndarray of Affine object
+
+    Parameters
+    ----------
+    transform: ndarray (Affine)
+
+    Returns
+    -------
+    - : ndarray (3x3)
+        The equivalent ndarray
+    """
+    return np.fromiter(transform, dtype='float64').reshape(3, 3)
+
+
+def translate_ndarray(transform, xoff, yoff):
+    """
+    Translate ndarray representing an affine transformation
+
+    Parameters
+    ----------
+    transform: ndarray (3x3)
+    xoff: number
+    yoff: number
+
+    Returns
+    -------
+    translated: ndarray
+        A transform with the translation applied.
+    """
+    translated = transform.copy()
+    translated[:, 2] = transform @ np.array([xoff, yoff, 1])
+    return translated
+
+def rowcol_np(transform, xs, ys, op='floor', precision=None):
```

Comment:
@groutr I'm not in favor of new methods, only in what numpy can do for us under the hood.

I experimented a bit last night and could only measure performance gains when inputs and outputs are numpy arrays instead of Python lists. If we changed the output type of, say, `rowcol` to be a sequence of sequences of int, we'd have a standard interface that a tuple of lists and a numpy array would both support, and we could do concrete type coercion according to the inputs for backwards compatibility. But that would be better left to rasterio 1.3.0, I think.

Reply:
Sorry about being unclear about my preference before.